### DISCUSSION TITLE
Creating multiple Application projects in Clean Architecture

### DISCUSSION THREAD

**Discussion by JunaidSultan20** [jasontaylordev/CleanArchitecture#577] [Q&A]
Can we create multiple Application projects/layers in the Clean Architecture.
For example I have a application project named Application.Inventory and now I want to add another application project named Application.HRM.
So is it okay if I add multiple application layers/projects in a clean architecture solution?

**Comment by ronnieholm** [comment]
This sounds like multiple DDD bounded contexts which would typically be indepedant systems. 

What benefit to you would adding multiple Application projects give you over multiple independant systems? Try listing the pros/cons so we can better understand your reasoning.

**Comment by Leanwit** [comment]
I agree with Ronnie, sounds like two bounded contexts.
I've seen some repositories where you can find more than one of bounded contexts.
In that case, they have the following structure:

src/
├─ Inventory/
│  ├─ Inventory.Application/
│  ├─ Inventory.Domain/
│  ├─ Inventory.Infrastructure/
├─ HRM/
│  ├─ HRM.Application/
│  ├─ HRM.Domain/
│  ├─ HRM.Infrastructure/

You won't be able to do imports between the bounded contexts if you are following the DDD approach, so in that case, you should create a Share folder with their respective Application, Domain, and Infrastructure folders. 

> Try listing the pros/cons so we can better understand your reasoning.

It would be great to know more about this to check if we can provide you with a better answer

**Comment by rene-titulaer** [comment]
Hi all,

I have the same question. I have several background projects.  Especially when you do a lot of async work you will end up with multiple background jobs handling the async work.

I also have two UI applications. 
One UI application is for customers and one is for the backoffice. It's the same domain.
 
I also have a web client and a mobile client, these are also different projects.

So a lot of deployable components. I used to call a deployable component an application but with clean architecture the term applications is arbitrary. I should call these deployable components presentations I assume, background processes are then presentations also.

**Comment by mweel1** [comment]
I did this by duplicating my application project for each application, and took common out of application and put it into its own project, and referenced it accordingly.

These applications still used domain, and infrastructure.
